go/libraries/doltcore/merge/action.go

```diff
@@ -98,26 +100,40 @@ func NewMergeSpec(ctx context.Context, rsr env.RepoStateReader, ddb *doltdb.Dolt
 		Squash:          squash,
 		Msg:             msg,
 		Noff:            noff,
+		NoCommit:        noCommit,
+		NoEdit:          noEdit,
 		Force:           force,
 		Email:           email,
 		Name:            name,
 		Date:            date,
 	}, true, nil
 }
 
+// MergeCommitSpec returns MergeStats after performing merge and commits if applied.
 // TODO forcing a commit with a constrain violation should warn users that subsequest
 // FF merges will not surface constraint violations on their own; constraint verify --all
 // is required to reify violations.
-func MergeCommitSpec(ctx context.Context, dEnv *env.DoltEnv, spec *MergeSpec) (map[string]*MergeStats, error) {
+func MergeCommitSpec(ctx context.Context, dEnv *env.DoltEnv, spec *MergeSpec) (bool, map[string]*MergeStats, error) {
+	var tblStats map[string]*MergeStats
 	if ok, err := spec.HeadC.CanFastForwardTo(ctx, spec.MergeC); err != nil && !errors.Is(err, doltdb.ErrUpToDate) {
-		return nil, err
+		return false, nil, err
 	} else if ok {
 		if spec.Noff {
-			return ExecNoFFMerge(ctx, dEnv, spec)
+			tblStats, err = ExecNoFFMerge(ctx, dEnv, spec)
+			return false, tblStats, err
 		}
-		return nil, ExecuteFFMerge(ctx, dEnv, spec)
+		return false, nil, ExecuteFFMerge(ctx, dEnv, spec)
 	}
-	return ExecuteMerge(ctx, dEnv, spec)
+	tblStats, err := ExecuteMerge(ctx, dEnv, spec)
+	if err != nil {
+		return false, tblStats, err
+	}
+
+	// merge will commit if
```

Comment:
These seem like they'd be more useful in the function godocs– great explanation for the different behavior and how the different flags affect it.